source/grid/tria.cc

```diff
@@ -14213,23 +14213,45 @@ typename Triangulation<dim, spacedim>::cell_iterator
   Triangulation<dim, spacedim>::create_cell_iterator(
     const CellId &cell_id) const
 {
+  Assert(this->contains_cell(cell_id),
+         ExcMessage(
+           "CellId is invalid for this triangulation.\n"
+           "Either the provided CellId does not correspond to a cell in this "
+           "triangulation object, or, in case you are using a parallel "
+           "triangulation, may correspond to an artificial cell that is less "
+           "refined on this processor."));
```

Comment:
Given the fullydistributed triangulation class, I suggest to augment the exception message to also describe that case.